scripts/lua/lib/dataStore.lua

```diff
@@ -0,0 +1,107 @@
+
+
+
+_M = {} 
+-- this script dynamically loads a persistence plugin and uses it to make calls to the persistent datastore
+
+
+local DATASTORE = os.getenv( 'DATASTORE')
+local utils = require('lib/utils')
+
+if DATASTORE == nil then 
+  DATASTORE = 'redis'
+end
+
+local impl = require(utils.concatStrings({'lib/', DATASTORE}))
+
+function _M.init() 
+  return impl.init()
+end 
+
+function _M.close(ds)
```

Comment:
Instead of needing to pass the `ds` object back to the `dataStore` thing itself, can't we just make `ds` and instance of the DAO itself so that you can do things like `ds.close()`?